DatadogCore/Sources/Core/Upload/DataUploadWorker.swift

```diff
@@ -102,6 +109,7 @@ internal class DataUploadWorker: DataUploadWorkerType {
                 DD.logger.debug("💡 (\(self.featureName)) No upload. Batch to upload: \(batchLabel), System conditions: \(blockersForUpload.description)")
 
                 self.delay.increase()
+                self.backgroundTaskCoordinator?.endCurrentBackgroundTaskIfActive()
```

Comment:
question/ To what I remember, we discussed app terminations if `end()` is not called within 30 seconds after `begin()`, no? How do we assure it 🤔 💭?

Reply:
It was a problem of previous setup - for some reason calling UIKit's handler wasn't ending in a balanced state, which was causing watchdog to terminate the process.

Here it's simplified in a way that single feature hold maximum of on background task registration and it works perfectly fine. You can test by commenting this line and observing that app goes into suspended state after 30 seconds.